test/common/common/shared_memory_hash_set_test.cc

```diff
@@ -136,7 +153,28 @@ TEST_F(SharedMemoryHashSetTest, tooManyValues) {
   value->number = 314519;
   EXPECT_EQ(1, hash_set1.size());
   EXPECT_EQ(314519, hash_set1.get(keys[options_.capacity])->number);
-  EXPECT_TRUE(hash_set1.sanityCheck());
+  hash_set1.sanityCheck();
+}
+
+TEST_F(SharedMemoryHashSetTest, severalKeysZeroHash) {
+  setUp<TestValueZeroHash>();
+  SharedMemoryHashSet<TestValueZeroHash> hash_set1(options_, true, memory_.get());
+  hash_set1.insert("one").first->number = 1;
+  hash_set1.insert("two").first->number = 2;
+  hash_set1.insert("three").first->number = 3;
+  EXPECT_TRUE(hash_set1.remove("two"));
+  hash_set1.sanityCheck();
+  hash_set1.insert("four").first->number = 4;
+  hash_set1.sanityCheck();
+  EXPECT_FALSE(hash_set1.remove("two"));
+  hash_set1.sanityCheck();
+}
+
+TEST_F(SharedMemoryHashSetTest, sanityCheckZeroedMemory) {
```

Comment:
nit: Can we name this a "DeathTest" per gtest recommendations?

Reply:
done